ros2_control_demo_description/rrbot_description/urdf/rrbot_description.urdf.xacro

```diff
@@ -0,0 +1,204 @@
+          <origin xyz="0 0 1" rpy="0 0 0" />
+          <parent link="${prefix}link2"/>
+          <child link="${prefix}tool_link" />
+        </joint>
+
+        <!-- Tool Link -->
+        <link name="${prefix}tool_link">
+        </link>
+
+        <joint name="${prefix}hokuyo_joint" type="fixed">
+          <axis xyz="0 1 0" />
+          <origin xyz="0 0 ${height3 - axel_offset/2}" rpy="0 0 0"/>
+          <parent link="${prefix}link2"/>
+          <child link="${prefix}hokuyo_link"/>
+        </joint>
+
+        <!-- Hokuyo Laser -->
+        <link name="${prefix}hokuyo_link">
+          <collision>
+            <origin xyz="0 0 0" rpy="0 0 0"/>
+            <geometry>
+              <box size="0.1 0.1 0.1"/>
+            </geometry>
+          </collision>
+
+          <visual>
+            <origin xyz="0 0 0" rpy="0 0 0"/>
+            <geometry>
+              <mesh filename="package://rrbot_description/meshes/hokuyo.dae"/>
+            </geometry>
+          </visual>
+
+          <inertial>
+            <mass value="1e-5" />
+            <origin xyz="0 0 0" rpy="0 0 0"/>
+            <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
+          </inertial>
+        </link>
+
+        <joint name="${prefix}camera_joint" type="fixed">
+          <axis xyz="0 1 0" />
+          <origin xyz="${camera_link} 0 ${height3 - axel_offset*2}" rpy="0 0 0"/>
+          <parent link="${prefix}link2"/>
+          <child link="${prefix}camera_link"/>
+        </joint>
+
+        <!-- Camera -->
+        <link name="${prefix}camera_link">
+          <collision>
+            <origin xyz="0 0 0" rpy="0 0 0"/>
+            <geometry>
+              <box size="${camera_link} ${camera_link} ${camera_link}"/>
+            </geometry>
+          </collision>
+
+          <visual>
+            <origin xyz="0 0 0" rpy="0 0 0"/>
+            <geometry>
+              <box size="${camera_link} ${camera_link} ${camera_link}"/>
+            </geometry>
+            <material name="red"/>
+          </visual>
+
+          <inertial>
+            <mass value="1e-5" />
+            <origin xyz="0 0 0" rpy="0 0 0"/>
+            <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
+          </inertial>
+        </link>
+
+        <!-- generate an optical frame http://www.ros.org/reps/rep-0103.html#suffix-frames
+        so that ros and opencv can operate on the camera frame correctly -->
+        <joint name="${prefix}camera_optical_joint" type="fixed">
+          <!-- these values have to be these values otherwise the gazebo camera image
+          won't be aligned properly with the frame it is supposedly originating from -->
+          <origin xyz="0 0 0" rpy="${-pi/2} 0 ${-pi/2}"/>
+          <parent link="${prefix}camera_link"/>
+          <child link="${prefix}camera_link_optical"/>
+        </joint>
+
+        <link name="${prefix}camera_link_optical">
+        </link>
+
+  </xacro:macro>
+
+</robot>
```

Comment:
I found `rrbot_description.urdf.xacro` and `rrbot.urdf.xacro` a bit confusing as we are already in the package `rrbot_description`. Could we not name `rrbot_description.urdf.xacro` `rrbot.kinematics_inertial_tree.xacro` ? I do not have a strong opinion on the naming.

Reply:
But it is much better than before.